backend/api/CourseApi/index.js

```diff
@@ -16,10 +16,20 @@ router.get('/public', catchAsync(async (request, response) => {
     offset: (pageNumber - 1) * pageSize,
     courseCategoryId: request.query.courseCategoryId
   });
-  const amountOfAllCourses = await CourseModel.select.countAllPublic({
-    courseCategoryId: request.query.courseCategoryId
-  });
 
+  var amountOfAllCourses;
+  var subquery = knex('problem').count('id').whereRaw('problem.course_id = course.id');
```

Comment:
It's ES6, let's use `const` for all vars that do  not change, and `let` for all vars that do get reassigned.
Fyi, - `const` and `let` are preferable, because they let us have `{}` scope instead of `function(){}` scope, - and that lets us use them easily within loops e.g..

Reply:
Sure, I knew I had to change this, but sent it for the functional check. Thats a good learning.